src/java/org/apache/cassandra/io/util/RandomAccessReader.java

```diff
@@ -85,8 +86,24 @@ public void readFloatsAt(long position, float[] dest) throws IOException
     {
         assert position % Float.BYTES == 0 : "Position must be aligned to multiple of " + Float.BYTES;
         var bh = rebufferer.rebuffer(position);
-        var floatBuffer = bh.floatBuffer();
-        floatBuffer.position(Ints.checkedCast((position - bh.offset()) / Float.BYTES));
+
+        FloatBuffer floatBuffer;
+        if (bh.offset() == 0)
```

Comment:
This is the exceptional case where we are positioned at the beginning of the file?

Reply:
where the *region* is positioned at the beginning of the file, yes